modules/nw-ovn-kubernetes-live-migration.adoc

```diff
@@ -0,0 +1,64 @@
+
+.Prerequisites
+
+* A cluster has been configured with the OpenShift SDN CNI network plugin in the network policy isolation mode.
+* You have installed the OpenShift CLI (`oc`).
+* You have access to the cluster as a user with the `cluster-admin` role.
+* You have created a recent backup of the etcd database is available.
+* The cluster is in a known good state without any errors.
+* On all cloud platforms after updating software, a security group rule must be in place to allow UDP packets on port `6081` for all nodes.
+* Cluster administrators must remove any egress router resources.
+* You have reviewed the "Considerations for live migration to the OVN-Kubernetes network plugin" section of this document.
+
+.Procedure
+
+. Before initiating the live migration, you must remove any egress router resources. If there is an egress router on the cluster when performing a live migration, the Network Operator blocks the migration and returns the following error: 
++
+[source,text]
+----
+The cluster configuration is invalid (network type live migration is not supported for pods with `pod.network.openshift.io/assign-macvlan` annotation. Please remove all egress router pods). Use `oc edit network.config.openshift.io cluster` to fix.
+----
++
+If this error is returned, you can query metrics on the {product-title} web console or by using the `oc` CLI. For more information, see "Checking live migration metrics".
+
+ifdef::openshift-rosa,openshift-dedicated[]
+. Enter the following command to add the `unsupported-red-hat-internal-testing` annotation to the cluster-level network configuration:
++
+[source,terminal]
+----
+$ oc patch Network.config.openshift.io cluster --type='merge' --patch '{"metadata":{"annotations":{"unsupported-red-hat-internal-testing": "true"}}}'
+----
+endif::[]
+
+. Enter the following command to patch the cluster-level networking configuration and initiate the migration from OpenShift SDN to OVN-Kubernetes:
++
+[source,terminal]
+----
+$ oc patch Network.config.openshift.io cluster --type='merge' --patch '{"metadata":{"annotations":{"network.openshift.io/network-type-migration":""}},"spec":{"networkType":"OVNKubernetes"}}'
+----
++
+After running these commands, the migration process begins. During this process, two rounds of node reboot are triggered with the Machine Config Operator. The expected time that this process takes is approximately two times that of a cluster upgrade.
+
+. Optional: You can enter the following commands to ensure that the migration process has completed, and to check the status of the `network.config`:
++
+[source,terminal]
+----
+$ oc get network.config.openshift.io cluster -o jsonpath='{.status.networkType}'
+----
++
+[source,terminal]
+----
+$ oc get network.config cluster -o=jsonpath='{.status.conditions}' | jq .
+----
++
+You can check live migration metrics for troubleshooting issues. For more information, see "Checking live migration metrics".
```

Comment:
Once, the live migration is completed. Users shall remove the annotation `network.openshift.io/network-type-migration` from the network.config CR.

Reply:
@pliurh added the following:

. Complete the following steps only if the migration succeeds and your cluster is in a good state:

.. To remove the migration configuration from the `network.config` custom resource, enter the following command:
+
[source,terminal]
----
$ oc patch Network.operator.openshift.io cluster --type='merge' \
  --patch '{ "spec": { "migration": null } }'
----

.. To remove custom configuration for the OpenShift SDN network provider, enter the following command:
+
[source,terminal]
----
$ oc patch Network.operator.openshift.io cluster --type='merge' \
  --patch '{ "spec": { "defaultNetwork": { "openshiftSDNConfig": null } } }'
----

.. To remove the OpenShift SDN network provider namespace, enter the following command:
+
[source,terminal]
----
$ oc delete namespace openshift-sdn
----